site/en/docs/extensions/reference/action/index.md

```diff
@@ -2,9 +2,165 @@
+choose to pin your extension icon to the toolbar.
+
+Note that every extension will have an icon in the toolbar in Chrome, even if
+the `action` key is not specified.
+
+## Manifest
+
+In order to use the `chrome.action` API, you need to specify the `action` key
+in your [manifest file][manifest].
+
+```json
+{
+  "name": "Action Extension",
+  "manifest_version": 3,
+  "action": {
+    "default_icon": {
+      "16": "images/icon16.png",
+      "24": "images/icon24.png",
+      "32": "images/icon32.png"
+    },
+    "default_title": "Click Me",
+    "default_popup": "popup.html"
+  },
+  ...
+}
+```
+
+Each of these values is optional; an empty dictionary is technically allowed.
+
+These properties are described more below.
+
+## Parts of the UI
+
+### Icon
+
+The icon is the main image used in the toolbar button. Icons are 16 DIPs
+(device-independent pixels) wide and tall. The icon is initially set by the
+`default_icon` key in the `action` entry in the `manifest.json` file. This
+key is a dictionary of sizes to image paths. Chrome will use these icons to
+choose which image scale to use. If an exact match is not found, Chrome will
+select the closest available and scale it to fit the image. However, this
+scaling can cause the icon to lose detail or look fuzzy.
+
+Since devices with less-common scale factors like 1.5x or 1.2x are becoming
+more common, you are encouraged to provide multiple sizes for your icons. This
+also ensures that if the icon display size is ever changed, you don't need to
+do any more work to provide different icons.
+
+The icon can also be set programmatically using the `action.setIcon()` method.
+This can be used to specify a different image path or to provide a
+dynamically-generated icon using the [HTML canvas element][canvas], or, if
+setting from an extension service worker, the
+[offscreen canvas][offscreenCanvas] API.
+
+```js
+const canvas = new OffscreenCanvas(16, 16);
+const context = canvas.getContext('2d');
+context.clearRect(0, 0, 16, 16);
+context.fillStyle = '#00FF00';  // Green
+context.fillRect(0, 0, 16, 16);
+const imageData = context.getImageData(0, 0, 16, 16);
+chrome.action.setIcon({imageData: imageData}, () => { ... });
+```
+
+!!!.aside
+The `action.setIcon()` API is intended to set a static image. It should not
+be used to simulate animation.
+!!!
+
+#### Formats
+
+For packed extensions (installed from a .crx file), images can be in most
+formats that the Blink rendering engine can display, including PNG, JPEG,
+BMP, ICO, and others (SVG is not supported). Unpacked extensions must use
+images in the PNG format.
+
+### Tooltip (title)
+
+The tooltip, or title, is displayed when the user hovers the mouse of the
+extension's icon in the toolbar. It is also included in the accessible text
+spoken by screenreaders when the button is focused.
+
+The default tooltip is set from the `default_title` field in the `action`
+`manifest.json` file. It can also be set programmatically with the
```

Comment:
Suggestion:
...you can also set it programmatically using the...

Reply:
Done